src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,261 @@ public static class NUnitTestExtractorApp
-                        }  
+                            Console.WriteLine(CommandLine.Text.HelpText.AutoBuild(_options));
 
-                        if (directory!=null && !Directory.Exists(directory))
-                        {
-                            Directory.CreateDirectory(directory);
+                            Environment.Exit(0);
                         }
 
-                        if (!File.Exists(_options.Output))
+                        // _options.Output is an absolute path of output file for the test name list
+                        if (!string.IsNullOrEmpty(_options.Output))
                         {
-                            using (FileStream stream = File.Create(_options.Output)) { }
+                            CreateOutputFileParentDirectory();
                         }
-                    }
 
-                    if (_options.Level != null)
-                    {
-                        StreamWriter writer = null;
-                        level = ParseLevel(_options.Level);
-                        try
-                        {
-                            if (_options.Output == null)
-                            {
-                                writer = new StreamWriter(Console.OpenStandardOutput());
-                            }
-                            else
-                            {
-                                writer = new StreamWriter(_options.Output, append: true);
-                            }
-
-                            GetTests(_options.DLLs, level, writer);
-                        }
-                        finally
+                        foreach (var assembly in _options.DLLs)
                         {
-                            if (writer != null)
-                            {
-                                writer.Dispose();
-                                
-                            }
+                            GenerateInfoFromAssembly(assembly);
                         }
                     }
+                    catch (Exception e)
+                    {
+                        Console.Error.WriteLine(StringUtils.FormatInvariant("Failed to process: {0}", e.InnerException));
+
+                        Environment.Exit(-2);
+                    }
                 }
             }
         }
 
-        /// <summary>
-        /// Writes each DLL file path along with either a namespace, class or function which contains tests(dependent on level specified by user)
-        /// </summary>
-        /// <param name="writer"> StreamWriter used to write the dll path and data, either to a txt file or stdout </param>
-        /// <param name="dll"> The dll path which is written </param>
-        /// <param name="data"> The test information corrosponding to the dll file </param>
-        private static void WriteTestDllAndName(TextWriter writer, string dll, string data)
+        public static string ScopeLevel
```

Comment:
[Should] This property is never used anywhere.  There's one place where you use: `Level lv = ParseLevel(_options.Level);` but you could use `Level lv = ParseLevel(ScopeLevel);` instead.
Also, since this should be a read-only property, you could simplify it to this:
`public static string ScopeLevel => _options.Level;`